### ISSUE TITLE
Show popup or notification for OSM message or changeset comment

### ISSUE THREAD

**Issue by aawiseman** [openstreetmap/iD#8468]
Often new mappers don't realize they have gotten a message or a changeset comment, which can be a problem if new mappers are contributing during a mapathon or event and don't see the feedback they are receiving. 

I think it would be good to have a pop-up window, blinking notification or some other obvious visual showing that they just received a message or changeset comment. I think this could help with new mapper retention as well.

**Comment by mxdanger** [comment]
This specific issue request relates to [openstreetmap-website](https://github.com/openstreetmap/openstreetmap-website). iD is just an embedded iframe inside of the OSM website that only appears while editing.

**Comment by aawiseman** [comment]
Right, I am aware that iD is a browser based editor, but it might be easier or better to implement as something that happens while editing live in iD rather than just on osm.org . For example, JOSM has a pop up type notification that you got a message as well which you only see when you are using JOSM. 

I would like something similar, that people to get an obvious notification while they are editing in iD, for example at a mapathon as I mentioned -- in addition to the little number that is next to your name as your browse OSM via a browser, outside of editing. I think more ways to make people aware that they are getting messages is a good thing.

**Comment by aawiseman** [comment]
In my experience the little number isn't obvious enough for new mappers -- I have personally sent a lot of messages and left many changeset comments for new mappers who aren't aware they are receiving them and don't know where or how to check.

**Comment by mxdanger** [comment]
> Right, I am aware that iD is a browser based editor, but it might be easier or better to implement as something that happens while editing live in iD rather than just on osm.org

Well, the thing is what you're suggesting wouldn't solve the issue and it would actually be harder and worse to implement. Adding a custom-made notification feature that is only seen inside of the iD editor iframe is not a great idea. Like I mentioned before, it would be far better to add the notification functionality to the openstreetmap-website and set up the Service Worker and Web Push Notification API to get notifications no matter where you are such as while you edit, view the map, or even have the tab closed.

**Comment by aawiseman** [comment]
I think it would solve the problem, the vast majority of new mappers use iD, and I bet it would be more likely to come to fruition in iD rather than OSM.org because I think not many changes happen to OSM.org. That said I will file it there too, but don’t expect much support.

**Comment by mxdanger** [comment]
> the vast majority of new mappers use iD

You have a misunderstanding. iD is just the editor that that appears inside of the OSM website when you click edit. You never leave the OSM website to edit with iD. The majority of users who use iD are on the OSM website. _Making edits with iD_, replying and viewing changesets/notes, sending messages, and etc are all done inside of the OSM website.

It doesn't make a lot of sense making custom JS notifications just for iD when you can have that handled by the OSM website with push notifications that appear natively in Android, Windows, and macOS for example.

**Comment by mxdanger** [comment]
Additionally, I don't even believe it's possible for iD to make a notification subscribe promise from inside the iframe.

**Comment by aawiseman** [comment]
Again, I know what iD is. It’s also embedded in the HOT Tasking Manager, so folks won’t necessarily be at OSM.org while editing. And in any case I still think this would be more likely to accepted as part of iD itself rather than as part of the OSM website. That said I would appreciate your support on the issue once I post it there.

**Comment by aawiseman** [comment]
I also raised this on the OSM website repo https://github.com/openstreetmap/openstreetmap-website/issues/3182 @mxdanger

**Comment by nickrsan** [comment]
Hi there - I realize this issue is a few years old now, and I'm not a maintainer, but it was resubmitted again by someone who also posted to [openstreetmap-website](https://github.com/openstreetmap/openstreetmap-website/issues/4080). The discussion there I think suggests reasons why this isn't a good task for iD yet, at least until the OSM website has additional features around tracking and managing notifications. I outlined a response in #9745 as to why I think this is likely going to be a `wontfix`, at least for now, even if maintainers weigh in (I'm just here to triage issues).

The short version is:
>I think gravitystorm's point in openstreetmap-website#4080 about building a proper notification system before solving this type of comment visibility issue is a good one - without that, iD would also struggle to know which notifications to surface (and wouldn't have a backing data store to record that information on its own), and would have to build out infrastructure that doesn't exist now, creating more fragmentation.

I'm marking this as wontfix, but leaving it open so that a maintainer can hopefully comment at some point.

**Comment by aawiseman** [comment]
@nickrsan There is a more active conversation in this issue https://github.com/openstreetmap/openstreetmap-website/issues/3182

**Comment by k-yle** [comment]
closing per the comments above
